_projects/coala_language_server.md

```diff
@@ -42,48 +42,110 @@ a rapidly growing list of editor and language communities:
 We have a state-of-art implementation in
 [coala-vs-code](https://github.com/coala/coala-vs-code) but it is designed
 for VS Code only. We need to design and implement a general mechanism for
-multiple editors/IDEs support.
+multiple editors/IDEs support. We also need clients written to support coala
+langserver for most if not all the supported editors. This is not because of
+the server's inability, but rather the need to configure the front end of the
+editor plugin such as defining how to start the server, communicate with it etc.
 
-The project is divided into two parts:
+The project is divided into three parts:
 
-1. Offer a basic languager server.
+1. Offer a basic language server.
 2. Implement a plugin based on the language server.
+3. Optimize the coala entry point and language server.
 
 #### Features
 
 - Support linting for multiple programming languages.
-- Support 'textDocument/publishDiagnostics', 'textDocument/didChange' and 'textDocument/didSave' requests.
+- The current language server has implementations for:
+  - 'textDocument/didSave'
+  - 'textDocument/didChange'
+  - 'textDocument/publishDiagnostics'
+  and this project will improve and extend the support to at least
+  the following request types:
+  - 'textDocument/didOpen'
+  - 'textDocument/didClose'
+  - 'workspace/formatting'
+  - 'window/showMessageRequest'
 - (Advanced) Multiple plugins Support (e.g. Eclipse Che).
+- Some additional request types that are either out of the scope
+  of this project or are marginally redundant to build but which would
+  improve the user friendliness or functionality  of this language
+  server are:
+  - 'textDocument/willSaveWaitUntil'
+  - 'workspace/didChangeConfiguration'
+  - 'workspace/configuration'
+  - 'workspace/applyEdit'
+  - 'window/showMessage'
```

Comment:
There's an odd mixture of client->server and server->client messages here. I'm not sure what it means for the Coala LS to "support" a server->client message.

Reply:
Some bears in coala can suggest changes to make the code confirm to the styling rules. It would be interesting if this functionality could also be provided under the same language server and make use of coala to its full potential.

- workspace/didChangeConfiguration: Control some of the coala configuration via editor settings.

- workspace/applyEdit: Apply workspace wide edits suggested by coala (fix the entire project).

- window/showMessage: Communicate messages raised by language server such as a missing coafile etc to the editor.

Reply:
My idea was that coala ls should support the second type of code fixing as you suggested, although the first way is better and user friendly. Stale suggestion handling could be a problem due to infrequent cycles of diagnostics being published. I think we can trade this for better overall experience. I will add ```workspace/executeCommand``` and ```textDocument/codeAction``` to the list.